build.sh

```diff
@@ -31,7 +31,7 @@ change_java_version() {
   fi
   echo "  JAVA_HOME=$JAVA_HOME"
   echo "  PATH=$PATH"
-  java -version
+  java -version || true
```

Comment:
Unfortunately Apple has seen fit to make this kind of solution a chore for either the developer or the user. If you're using macOS and you don't have java installed, you still have /usr/bin/java. When you invoke it, you get this lovely dialog:
 
<img width="488" alt="Screen Shot 2019-12-09 at 8 20 09 PM" src="https://user-images.githubusercontent.com/1566303/70486857-537e0700-1ac1-11ea-861c-214054639241.png">

This happening during `./build.sh docker` is undesirable. One solution would be to generalize my code from [#733](https://github.com/apache/avro/pull/733/files#diff-10c0faac6752a840e96aa1c99cd51740R67).

Reply:
I see!  Actually, that really isn't a bad thing...  does it block until a JDK is installed?  Maybe the right thing to do is just *not* call `change_java_version` until we're in a `build.sh` target that requires it.

In that case, if java is missing, `build.sh lint` would fail (the current behaviour once we get to the java language) or pop up the above dialog for MacOS and continue.  Idem for `test`, `clean` and `rat` (requires maven).

And `build.sh docker` would succeed (and `sign`, `githooks`, etc).

Reply:
It doesn't block, but it's still disruptive because it steals focus from the terminal.